### ISSUE TITLE
Save a page on [re]load?

### ISSUE THREAD

**Issue by brianjmurrell** [gildas-lormeau/single-file-cli#129]
Let's say I have a browser open and an [extension (for example)](https://chromewebstore.google.com/detail/easy-auto-refresh/aabcgdmkeabbnleenpncegpcngjpnjkc?hl=en) on a page is reloading it every 60 seconds.

Is there any way to get this extension to save the page on each reload?

**Comment by gildas-lormeau** [comment]
Are you using SingleFile as an extension or as a CLI tool?

**Comment by brianjmurrell** [comment]
No, sorry.  Posted this in the wrong project (feel free to move this issue if you can).  I have taken note of _Auto-save_>_Autosave this tab_ though and that's almost exactly what I want.  Well, it's what I want and I can work with it as it is if need be.  :smile:

The only issue I have is all of the HTML metadata wrapping of the page's contents.  In my case, the page content-type is `application/json; charset=utf-8` and being able to save just the raw JSON without the:

```html
<html><!--
 Page saved with SingleFile 
 url: https://[redacted]/api/accounts/summary 
 saved date: Fri Oct 04 2024 14:45:19 GMT-0400 (Eastern Daylight Time)
--><meta charset=utf-8><meta name=color-scheme content="light dark"><meta name=referrer content=no-referrer><link type=image/x-icon rel="shortcut icon" href="…"><link rel=canonical href=https://[redacted]/api/accounts/summary><meta http-equiv=content-security-policy content="default-src 'none'; font-src 'self' data:; img-src 'self' data:; style-src 'unsafe-inline'; media-src 'self' data:; script-src 'unsafe-inline' data:; object-src 'self' data:; frame-src 'self' data:;"><style>img[src="data:,"],source[src="data:,"]{display:none!important}</style><body style=zoom:1><span id=warning-container></span><pre>{[redacted JSON]}</pre><div class=json-formatter-container></div>
```

wrapping would be most ideal.

**Comment by gildas-lormeau** [comment]
You could clean up the page by executing a userscript (see https://github.com/gildas-lormeau/SingleFile/wiki/How-to-execute-a-user-script-before-a-page-is-saved). In that case, maybe running `document.open(); document.write(warning-container); document.close();` could work.
